kornia/augmentation/param_gen.py

```diff
@@ -258,12 +258,46 @@ def _random_rotation_gen(batch_size: int, degrees: FloatUnionType) -> Dict[str,
     return params
 
 
+def _compose_affine_matrix_3x3(translations: torch.Tensor,
+                               center: torch.Tensor,
+                               scale: torch.Tensor,
+                               angle: torch.Tensor,
+                               sx: Optional[torch.Tensor],
+                               sy: Optional[torch.Tensor]) -> torch.Tensor:
+    r"""Composes affine matrix Bx3x3 from the components
+    Returns:
+        torch.Tensor: params to be passed to the affine transformation.
+    """
+    if sx is not None:
+        x, y = torch.split(center, 1, dim=-1)
+        x = x.view(-1)
+        y = y.view(-1)
+        sx_tan = torch.tan(sx)  # type: ignore
+        sy_tan = torch.tan(sy)  # type: ignore
+        zeros = torch.zeros_like(sx)  # type: ignore
+        ones = torch.ones_like(sx)  # type: ignore
+        shear_mat = torch.stack([ones,   -sx_tan,                 sx_tan * x,  # type: ignore   # noqa: E241
+                                 -sy_tan, ones + sx_tan * sy_tan, sy_tan * (-sx_tan * x + y)],  # noqa: E241
+                                dim=-1).view(-1, 2, 3)
```

Comment:
maybe we could provide a functionality like `get_shear_matrid2d` similar to rotation

Reply:
Good idea, but I am not sure how to handle translation there. We have a special case here for being around center. The general case is not clear to me...

Reply:
![image](https://user-images.githubusercontent.com/15955486/89497604-413e9f00-d7ef-11ea-8442-11b8e686ddc7.png)
@ducha-aiki I am trying to implement the 3D version but I cannot reproduce this translation here. Can you let me know the exact math you did?

Reply:
@shijianjian you are right, good catch. I messed up x and y somehow